src/software/new_geom/convex_shape.h

```diff
@@ -3,6 +3,12 @@
 /**
  * A shape that is convex (curved outwards).
  */
-class ConvexShape : public Shape
+class ConvexShape : public virtual Shape
```

Comment:
I don't think you need the `virtual` when inheriting like this

Reply:
For context, checkout "The solution" on this page for an example of why you might want to do virtual inheritance like this: https://en.wikipedia.org/wiki/Virtual_inheritance